launcher/BaseInstance.cpp

```diff
@@ -369,6 +365,13 @@ shared_qobject_ptr<LaunchTask> BaseInstance::getLaunchTask()
     return m_launchProcess;
 }
 
+void BaseInstance::setSetting(const QString& key, const QVariant& value)
```

Comment:
wouldn't this check make more sense in `Setting::set` / `SettingsObject::set` instead? this seems like something we'd like to do in other places too. To avoid breaking weird behavior throughout the codebase (maybe someone relies on it always emitting the signal for e.g.  having a tristate and setting the value to the default one but still saving it to the file), there could be a bool argument to force or not saving, defaulting to do so.

Reply:
agree